pycfmodel/model/resources/resource.py

```diff
@@ -43,4 +45,28 @@ def has_hardcoded_credentials(self) -> bool:
 
     @property
     def policy_documents(self) -> List[OptionallyNamedPolicyDocument]:
-        return []
+        """
+        Returns a list with all the optionally named policy documents in this resource within its properties.
+        Every resource has a Properties field, if not, it's a malformed CloudFormation template.
+        """
+        policy_documents = []
+        self.obtain_policy_documents(
+            policy_documents=policy_documents, properties=list(self.Properties.__dict__.values())
```

Comment:
`.dict()` returns a dictionary of the model's fields and values

```suggestion
            policy_documents=policy_documents, properties=list(self.Properties.dict().values())
```

Reply:
tested it locally both for `dict()` and `dir()` and tests don't pass.
When using `dict()`, the value given to the function is a list of python types not mapped to models, that is, a PolicyDocument is mapped to a dictionary instead of a PolicyDocument

Reply:
ahh, yes because it's recursive.
Can you try with `_iter()`?